packages/firestore/src/local/memory_remote_document_cache.ts

```diff
@@ -235,34 +194,29 @@ export class MemoryRemoteDocumentChangeBuffer extends RemoteDocumentChangeBuffer
   protected applyChanges(
     transaction: PersistenceTransaction
   ): PersistencePromise<void> {
-    const changes = this.assertChanges();
-    let delta = 0;
-    const docs: DocumentSizeEntry[] = [];
-    changes.forEach((key, maybeDocument) => {
-      const previousSize = this.documentSizes.get(key);
-      assert(
-        previousSize !== undefined,
-        `Attempting to change document ${key.toString()} without having read it first`
-      );
-      const size = this.sizer(maybeDocument);
-      delta += size - previousSize!;
-      docs.push({ maybeDocument, size });
+    const promises: Array<PersistencePromise<void>> = [];
+    this.changes.forEach((key, doc) => {
+      if (doc) {
+        const size = this.sizer(doc);
+        promises.push(this.documentCache.addEntry(transaction, doc, size));
```

Comment:
Hrm...  Maybe not a big deal, but it seems like you could move `this.sizer(doc)` into the implementation of `addEntry()` and then IndexedDbRemoteDocumentCache.addEntry() and MemoryRemoteDocumentCache.addEntry() would have similar signature which seems appealing.

Reply:
Done. It keeps the size tracking even more encapsulated, which is something I am in favor of.